Fix OSC next/prev media buttons skip files, #5611

low-batt opened the pull request:
This commit will:
- Rename the `updateArrowButtonImage` method in `MainWindowController` to be `updateArrowButtons`
- Change `updateArrowButtons` to configure the button type to be `momentaryPushIn` or `multiLevelAccelerator` based on what the OSC buttons are set to do
- Change the `leftButtonAction` and `rightButtonAction` methods to handle `momentaryPushIn` buttons

With these changes the OSC left and right arrow buttons only change the position in the playlist by one file. This also corrects the behavior of the buttons when set to seek to only seek once.

This also corrects the behavior where a pressure sensitive device was providing tactile feedback when the buttons were configured to change the position in the playlist, or seek forward. As the buttons only take into account the pressure being applied when the buttons are set to control playback speed it was inappropriate to be providing pressure feedback when the buttons were not controlling the speed.

- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [x] This implements/fixes issue #5611.

---

**Description:**

low-batt committed:
Fix OSC next/prev media buttons skip files, #5611

This commit will:
- Rename the updateArrowButtonImage method in MainWindowController to be
  updateArrowButtons
- Change updateArrowButtons to configure the button type to be
  momentaryPushIn or multiLevelAccelerator based on what the OSC buttons
  are set to do
- Change the leftButtonAction and rightButtonAction methods to handle
  momentaryPushIn buttons

With these changes the OSC left and right arrow buttons only change the
position in the playlist by one file. This also corrects the behavior
of the buttons when set to seek to only seek once.

This also corrects the behavior where a pressure sensitive device was
providing tactile feedback when the buttons were configured to change
the position in the playlist, or seek forward. As the buttons only take
into account the pressure being applied when the buttons are set to
control playback speed it was inappropriate to be providing pressure
feedback when the buttons were not controlling the speed.

low-batt commented:
See [this comment](https://github.com/iina/iina/issues/5611#issuecomment-3420031030) for an analysis of what is going wrong.

uiryuu commented on iina/MainWindowController.swift:
```diff
@@ -2897,16 +2923,24 @@ class MainWindowController: PlayerWindowController {
+  /// - Parameter sender: The button invoking this action.
   @IBAction func rightButtonAction(_ sender: NSButton) {
-    if arrowBtnFunction == .speed {
+    log("$$$$$$$$ \(sender)")
```
Maybe remove this before merging?

low-batt replied:
Embarrassing. Good catch! Fixed.

low-batt committed:
Fix OSC next/prev media buttons skip files, #5611

This commit will remove a log message added for debugging that was
accidentally left in the code.

Balamiere commented:
Can we please release this fix into a general release? It's been very annoying not being able to skip to a next episode properly. Thank you.

lhc70000 commented:
We are planning a release in several days.

Balamiere commented:
> We are planning a release in several days.

yay! big thank you! can't wait for the release :)

Balamiere commented:
It's been more than a week. Home many days is "several days"? Do we have a tentative date for planned release?

uiryuu commented:
We're doing some final tests and some final adjustments. We hope we can release in a day or so. Please be patient and stay tuned.

lhc70000 commented:
We had a build last weekend but found some issues needing urgent fix (see the current develop branch). We are releasing it in hours.

Balamiere commented:
I can see 1.4.2 is released now! Yay! Thank you, thank you, thank you!
